lib/rex/post/meterpreter/extensions/stdapi/fs/dir.rb

```diff
@@ -240,6 +240,21 @@ def Dir.rmdir(path)
     delete(path)
   end
 
+  #
+  # Returns true if the remote directory exists, false otherwise.
+  #
+  def Dir.exist?(path)
+    begin
+      entries(path)
+    rescue ::Rex::Post::Meterpreter::RequestError => e
+      if e.message =~ /The system cannot find the path specified/
+        return false
+      else
+        raise e
+      end
+    end
+  end
```

Comment:
There is a built in function to do what you are trying to do here which is located at https://github.com/rapid7/metasploit-framework/blob/b7d8fb1ee4fdad4e41725e3f24c47c23e5c3f14a/lib/msf/core/post/file.rb#L228-L246 for files and for directories you can use the code at https://github.com/rapid7/metasploit-framework/blob/b7d8fb1ee4fdad4e41725e3f24c47c23e5c3f14a/lib/msf/core/post/file.rb#L98-L117. It would be better to use these functions as they are already built into the framework and this way if anything breaks we don't have to fix multiple custom functions and can instead fix the one function that all the modules rely on.

Reply:
@cn-kali-team To be honest now that I look at this I do recall that I think this function was broken somewhat on Meterpreter but worked fine on normal non-Meterpreter Windows and Linux sessions. I might be a better idea to look into perhaps changing the existing function to address this issue, but if we are going to do this it should be done in a separate PR. This PR should only contain changes related to the FoxMail password gatherer. It should not contain any library related changes.

Reply:
Changes to this library have been undone, so marking this as resolved.